internal/cluster/distribution/pke/service.go

```diff
@@ -120,17 +124,28 @@ type service struct {
 	genericClusters Store
 	clusterManager  ClusterManager
 	nodePoolManager NodePoolManager
+	nodePoolStore   NodePoolStore
 }
 
 // +testify:mock:testOnly=true
 
 // NodePoolManager is responsible for managing node pools.
 type NodePoolManager interface {
-	// UpdateNodePool updates an existing node pool in a cluster.
-	UpdateNodePool(ctx context.Context, c cluster.Cluster, nodePoolName string, nodePoolUpdate NodePoolUpdate) (string, error)
+	// DeleteNodePool deletes an existing node pool in a cluster.
+	DeleteNodePool(
+		ctx context.Context,
+		c cluster.Cluster,
+		existingNodePool ExistingNodePool,
+		shouldUpdateClusterStatus bool,
+	) (err error)
 
-	// ListNodePools lists node pools from a cluster.
-	ListNodePools(ctx context.Context, c cluster.Cluster, nodePoolNames []string) ([]NodePool, error)
+	// UpdateNodePool updates an existing node pool in a cluster.
+	UpdateNodePool(
+		ctx context.Context,
+		c cluster.Cluster,
+		nodePoolName string,
+		nodePoolUpdate NodePoolUpdate,
+	) (string, error)
```

Comment:
**Question**: why is the `NodePoolManager.UpdateNodePool` addition part of this PR?
Does `DeleteNodePool` use it anywhere (it should not IMO)?

Reply:
It's not an addition, I change the order and formatted the code.